### DISCUSSION TITLE
How can I test schedule execution?

### DISCUSSION THREAD

**Discussion by NakulK48** [dagster-io/dagster#4004] [Q&A] [answered]
This seems to be the missing piece in testing pipelines, and therefore it's invariably what goes wrong when we try to deploy - stuff like forgetting the dummy argument in the `schedule` function, or a logic error when loading the config for the run.

The function itself is decorated and there's no easy way of getting the function object back. `schedule._run_config_fn` raises an attribute error, `schedule.get_execution_data()` needs a `context` and nothing else in `dir(schedule)` seems relevant.

Creating a dummy `context` (we don't actually use that argument for anything) doesn't work either - it complains with some esoteric error about partitions.

**Comment by NakulK48** [comment, marked as answer]
This appears to work:

```python
instance = DagsterInstance.get().get_ref()
context = ScheduleExecutionContext(instance, scheduled_execution_time=None)
[run_request] = schedule.get_execution_data(context)
```

**Reply by sryza** [reply]
@NakulK48 this is right.  We're currently working on documenting this and considering an API that would be a little less verbose.
